checks/evaluation/security_policy.go

```diff
@@ -20,137 +20,62 @@ import (
 	"github.com/ossf/scorecard/v4/finding"
 )
 
-func scoreSecurityCriteria(f checker.File,
-	info []checker.SecurityPolicyInformation,
-	dl checker.DetailLogger,
-) int {
-	var urls, emails, discvuls, linkedContentLen, score int
-
-	emails = countSecInfo(info, checker.SecurityPolicyInformationTypeEmail, true)
-	urls = countSecInfo(info, checker.SecurityPolicyInformationTypeLink, true)
-	discvuls = countSecInfo(info, checker.SecurityPolicyInformationTypeText, false)
-
-	for _, i := range findSecInfo(info, checker.SecurityPolicyInformationTypeEmail, true) {
-		linkedContentLen += len(i.InformationValue.Match)
-	}
-	for _, i := range findSecInfo(info, checker.SecurityPolicyInformationTypeLink, true) {
-		linkedContentLen += len(i.InformationValue.Match)
-	}
-
-	msg := checker.LogMessage{
-		Path: f.Path,
-		Type: f.Type,
-		Text: "",
-	}
-
-	// #1: linked content found (email/http): score += 6
-	if (urls + emails) > 0 {
-		score += 6
-		msg.Text = "Found linked content in security policy"
-		dl.Info(&msg)
-	} else {
-		msg.Text = "no email or URL found in security policy"
-		dl.Warn(&msg)
-	}
-
-	// #2: more bytes than the sum of the length of all the linked content found: score += 3
-	//     rationale: there appears to be information and context around those links
-	//     no credit if there is just a link to a site or an email address (those given above)
-	//     the test here is that each piece of linked content will likely contain a space
-	//     before and after the content (hence the two multiplier)
-	if f.FileSize > 1 && (f.FileSize > uint(linkedContentLen+((urls+emails)*2))) {
-		score += 3
-		msg.Text = "Found text in security policy"
-		dl.Info(&msg)
-	} else {
-		msg.Text = "No text (beyond any linked content) found in security policy"
-		dl.Warn(&msg)
+// SecurityPolicy applies the score policy for the Security-Policy check.
+func SecurityPolicy(name string, findings []finding.Finding) checker.CheckResult {
```

Comment:
It is. 

There's a way to hack around it and duplicate the code, but it will go out of sync. I'm personally in favor of accepting the breaking change unless you have a better idea. I've been arguing that all these internal functions should be made out of scope of semver... and put them under an internal folder. The only API we have semver on is the one on `pkg.RunScorecard`.

Let me know what you think. Are we OK delaying the next release until these changes land?

Reply:
Note that we broke backward compatibility in these functions severals time over the last few releases, e.g. https://github.com/ossf/scorecard/tree/v4.11.0/checks/evaluation/permissions and https://github.com/ossf/scorecard/tree/v4.11.0/probes.

I'm ready to update the README about semantic versioning.

Reply:
> I've been arguing that all these internal functions should be made out of scope of semver in the README... and put them under an internal folder.

If they're in an `internal` folder, then yes they're out of scope of semver. I am in favor of keeping as much as we need internal as possible.

Reply:
Note: Moving things internal (I think we should wait until v5), may negatively impact Allstar, so we should double check what public functionality we want to offer.

> I'm ready to update the README about semantic versioning.

Unfortunately the Go tool cares about our README on semver. It expects Go modules to follow it. I think this introduces some scenarios with Minimal Version Selection that could negatively impact some projects.

> Are we OK delaying the next release until these changes land?

I don't think we want to block the GitLab release

Reply:
> Moving things internal (I think we should wait until v5), may negatively impact Allstar, so we should double check what public functionality we want to offer.

afaik, allstar only uses pkg.RunScorecard, nothing else. @raghavkaul can you confirm?

> Unfortunately the Go tool cares about our README on semver. It expects Go modules to follow it. I think this introduces some scenarios with Minimal Version Selection that could negatively impact some projects.

The go compiler read the README?

Reply:
so checked allstar and it uses the check's API, but not the raw or evaluation API.